compiler-plugin-test/src/test/java/io/ballerina/stdlib/persist/sql/compiler/CompilerPluginTest.java

```diff
@@ -0,0 +1,419 @@
+                },
+                new String[]{
+                        "(20:4,21:29)",
+                        "(32:4,33:24)",
+                        "(43:4,44:23)"
+                }
+        );
+    }
+
+    @Test(enabled = true)
+    public void validateRelationAnnotations1() {
+        List<Diagnostic> diagnostics = getErrorDiagnostics("project_1",
+                "relation1.bal", 1);
+        testDiagnostic(
+                diagnostics,
+                new String[]{
+                        PERSIST_423.getCode()
+                },
+                new String[]{
+                        "invalid use of Relation annotation. mismatched number of reference " +
+                                "keys for entity 'Car' for relation 'Person'. expected 2 but found 1."
+                },
+                new String[]{
+                        "(19:4,20:17)"
+                }
+        );
+    }
+
+    @Test(enabled = true)
+    public void validateRelationAnnotations2() {
+        List<Diagnostic> diagnostics = getErrorDiagnostics("project_1",
+                "relation2.bal", 1);
+        testDiagnostic(
+                diagnostics,
+                new String[]{
+                        PERSIST_424.getCode()
+                },
+                new String[]{
+                        "invalid use of Relation annotation. mismatched key types for entity " +
+                                "'Person' for its relationship."
+                },
+                new String[]{
+                        "(18:4,19:17)"
+                }
+        );
+    }
+
+    @Test(enabled = true)
+    public void validateRelationAnnotations3() {
+        List<Diagnostic> diagnostics = getErrorDiagnostics("project_1",
+                "relation3.bal", 1);
+        testDiagnostic(
+                diagnostics,
+                new String[]{
+                        PERSIST_426.getCode()
+                },
+                new String[]{
+                        "invalid use of Relation annotation. the field 'cars' is an array type in a 1-n " +
+                                "relationship. therefore, it cannot have foreign keys."
+                },
+                new String[]{
+                        "(9:4,10:15)"
+                }
+        );
+    }
+    @Test(enabled = true)
+    public void validateRelationAnnotations4() {
+        List<Diagnostic> diagnostics = getErrorDiagnostics("project_1",
+                "relation4.bal", 1);
+        testDiagnostic(
+                diagnostics,
+                new String[]{
+                        PERSIST_427.getCode()
+                },
+                new String[]{
+                        "invalid use of Relation annotation. the field 'car' is an optional type in a 1-1 " +
+                                "relationship. therefore, it cannot have foreign keys."
+                },
+                new String[]{
+                        "(9:4,10:13)"
+                }
+        );
+    }
```

Comment:
Let's have a new line before the test functions. It's more readable in that way. Please check other places as well.